### DISCUSSION TITLE
How to configure Celery task logs in FastAPI

### DISCUSSION THREAD

**Discussion by upeshphable** [fastapi/fastapi#8580] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import logging
import os
from logging import handlers, Formatter
from pythonjsonlogger import jsonlogger

from app.config import settings
from celery import Celery
from celery.signals import after_setup_logger

celery = Celery(__name__)
celery.conf.broker_url = settings.CELERY_BROKER_URL or "redis://localhost:6379"
celery.conf.update(worker_hijack_root_logger=False)


@after_setup_logger.connect()
def logger_setup_handler(logger, **kwargs):
    celery_log_handler = handlers.TimedRotatingFileHandler(os.path.join(os.getenv("OPS_BASE_PATH"), "logs", "celery.log"), 'h', 24, 5)
    celery_log_handler.setLevel(logging.DEBUG)
    my_formatter = jsonlogger.JsonFormatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')  # custom formatter
    # my_formatter = Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')  # custom formatter
    celery_log_handler.setFormatter(my_formatter)

    logger.addHandler(celery_log_handler)

    logger.debug("Celery logger setup done")


class SomeTask(celery.Task):

    name = "Some Task Name"

    def _run(self, *args, **kwargs):

        logger.debug("Task started")

        asyncio.run(self.execute(args[0]))

    async def execute(self, fulfillment_id):

        logger = get_task_logger("app.jobs.periodic")
        print(f"Will expire the fulfillment id: {fulfillment_id}")
```

### Description

In the example code, "Task started" message from _run method get printed in the log. However, all log messages from within execute method don't get printed. When I used simple print method, it got printed in the log under "celery.redirected" with level WARNING.

Can someone please help me configure the logging.

### Operating System

macOS

### Operating System Details

macOS Monterey

### FastAPI Version

0.70.0

### Python Version

3.10.2

**Comment by k-aleksandrovich** [comment]
Same thing

**Comment by ShantanuKudva** [comment]
Are there any updates on this? I want to change the log handler to that of Loguru. I would love to know if there are any best practices to achieve this
